Behavior/Announce.cs

```diff
@@ -0,0 +1,131 @@
+﻿using Discord.WebSocket;
+using Reiati.ChillBot.Data;
+using Reiati.ChillBot.Tools;
+using System.Threading.Tasks;
+
+namespace Reiati.ChillBot.Behavior
+{
+    /// <summary>
+    /// Responsible for managing behavior related to making announcements.
+    /// </summary>
+    public class Announce
+    {
+        /// <summary>
+        /// Announces the creation of a channel in the given guild if an annoucement channel is configured.
+        /// </summary>
+        /// <param name="guild">Information about the guild. May not be null.</param>
+        /// <param name="requestAuthor">The author of the channel create request. May not be null.</param>
+        /// <param name="channelName">The name of the new channel. May not be null.</param>
+        /// <param name="channelDescription">The description of the new channel.</param>
+        /// <returns>A task indicating completion.</returns>
+        public static async Task ChannelCreation(
+            Guild guild,
+            SocketGuildUser requestAuthor,
+            string channelName,
+            string channelDescription)
+        {
+            ValidateArg.IsNotNullOrWhiteSpace(channelName, nameof(channelName));
+
+            await Announce.SendAnnouncementMessageAsync(
+                guildConnection: requestAuthor.Guild,
+                guild: guild,
+                message: $"<@{requestAuthor.Id}> created a new channel named **{channelName}** with the description \"{channelDescription}\"\n" +
```

Comment:
nit: If `channelDescription` is null or empty the message may be weird.
```csharp
string message = $"<@{requestAuthor.Id}> created a new channel named **{channelName}**";
if (!string.IsNullOrWhiteSpace(channelDescription))
{
   message += $" with the description \"{channelDescription}\"";
}
message += + $"\nLet me know if you're interested by using a command like, \"/join {channelName}\"")
```

Reply:
True. I believe all of the code paths that currently lead here validate the description is not null or empty, but it can't hurt to make this method more generic.